experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,619 @@
+    ev_gbasis_new = [mapp(ev_gbasis[i]) for i in 1:length(ev_gbasis)]
+    if length(ev_gbasis_new) == 0
+        return [mixed_ring_graded()]
+    end
+    return minimal_generating_set(ideal(ev_gbasis_new))
+end
+
+#now we have to perform reynolds operation. This will happen in mixed_ring_xy. 
+#the elements returned will be in the polynomial ring K[X]
+function inv_generators(I::MPolyIdeal, G::ReductiveGroup, ringg::MPolyRing, M::AbstractAlgebra.Generic.MatSpaceElem, X::MPolyIdeal, reynolds_function::Function)
+    genss = gens(I)
+    if length(genss) == 0
+        return Vector{elem_type(ringg)}()
+    end
+    mixed_ring_xy = parent(genss[1])
+    R = base_ring(M)
+    m = G.group[2]
+    n = ncols(M)
+    mapp = hom(R, mixed_ring_xy, gens(mixed_ring_xy))
+    new_rep_mat = matrix(mixed_ring_xy,n,n,[mapp(M[i,j]) for i in 1:n, j in 1:n])
+    #we need det_
+    det_ = det(G.canonical_representation)
+    mapp_ = hom(parent(det_), mixed_ring_xy, gens(mixed_ring_xy)[(2*n)+1:(2*n)+(m^2)])
+    new_det = mapp_(det_)
+    if G.group[1] == :SL #TODO other types of reductive groups
+        new_gens_wrong_ring = [reynolds_function(genss[i], new_rep_mat, new_det, m) for i in 1:length(genss)]
+    else 
+        return nothing
+    end
+    img_genss = vcat(gens(ringg), zeros(ringg, n+m^2))
+    mixed_to_ring = hom(mixed_ring_xy, ringg, img_genss)
+    new_gens = Vector{MPolyDecRingElem}(undef,0)
+    for elemm in new_gens_wrong_ring
+        if elemm != 0
+            push!(new_gens, mixed_to_ring(elemm))
+        end
+    end
+    if length(new_gens) == 0
+        return [ringg()]
+    end
+    
+    #remove ugly coefficients: 
+    V= Vector{FieldElem}[]
+    new_gens_ = Vector{MPolyDecRingElem}(undef,0)
+    for elem in new_gens
+        V = vcat(V,collect(coefficients(elem)))
+        maxx = maximum([abs(denominator(V[i])) for i in 1:length(V)])
+        minn = minimum([abs(numerator(V[i])) for i in 1:length(V)])
+        if denominator((maxx*elem)//minn) != 1
+            error("den not 1")
+        end
+        push!(new_gens_, numerator((maxx*elem)//minn))
+        V= Vector{FieldElem}[]
+    end
+    return new_gens_
+end
+
+
+function mu_star(new_rep_mat::AbstractAlgebra.Generic.MatSpaceElem)
+    mixed_ring_xy = parent(new_rep_mat[1,1])
+    n = ncols(new_rep_mat)
+    vars = matrix(mixed_ring_xy,1,n,[gens(mixed_ring_xy)[i] for i in 1:n])
+    new_vars = vars*transpose(new_rep_mat)
+    D = Dict([])
+    for i in 1:n
+        push!(D, gens(mixed_ring_xy)[i]=>new_vars[i])
+    end
+    return D
+end
+
+function reynolds_v_slm(elem::MPolyDecRingElem, new_rep_mat::AbstractAlgebra.Generic.MatSpaceElem, new_det::MPolyDecRingElem, m::Int)
+    D = mu_star(new_rep_mat)
+    mixed_ring_xy = parent(elem)
+    sum_ = mixed_ring_xy()
+    #mu_star: 
+    mons = collect(monomials(elem))
+    coeffs = collect(coefficients(elem))
+    for i in 1:length(mons)
+        k = mixed_ring_xy(1)
+        factors = factorise(mons[i])
```

Comment:
This use of `factorise` still should be removed. And the code above be rewritten to use `terms`.